clash-ghc/src-ghc/Clash/GHC/Evaluator/Primitive.hs

```diff
@@ -103,7 +103,8 @@ import {-# SOURCE #-} Clash.GHC.Evaluator
 
 isUndefinedPrimVal :: Value -> Bool
 isUndefinedPrimVal (PrimVal (PrimInfo{primName}) _ _) =
-  primName == "Clash.Transformations.undefined"
+  primName `elem` ["Clash.Transformations.undefined"
+                  ,"Clash.XException.errorX"]
```

Comment:
1. Can the evaluator access blackbox internals? If so, could we replicate the behavior in `Clash.Normalize.Transformations.isPrimError`? 

2. Should we add `"Clash.Transformations.removedArg"` too?

Reply:
1. No, it does not have access
2. No, definitely not... having `removedArg` be a subject of a case-expression would be a bug